stdlib/public/Volatile/Volatile.swift

```diff
@@ -0,0 +1,177 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Swift
+
+/// A pointer for accessing "volatile" memory, e.g. memory-mapped I/O registers.
+///
+/// Do not use for inter-thread synchronization. This is only meaningful for
+/// low-level operations on special memory addresses performed from OS kernels,
+/// embedded firmware, and similar environments.
+///
+/// The semantics of volatile load and volatile store operations match the LLVM
+/// volatile semantics. Notably, a volatile operation cannot be added, removed,
+/// or reordered with other volatile operations by the compiler. They may be
+/// reordered with non-volatile operations. For details, see
+/// <https://llvm.org/docs/LangRef.html#volatile-memory-accesses>.
+public struct UnsafeVolatilePointer<Pointee> {
+  public var _rawPointer: Builtin.RawPointer
+  public init(bitPattern: UInt) {
+    _rawPointer = UnsafeRawPointer(bitPattern: bitPattern)!._rawValue
```

Comment:
what happens when someone passes 0? this will assert - are there ever memory mapped registers at 0x0?

Should this not use `self.init(Builtin.inttoptr_Word(bitPattern._builtinWordValue))`?

Reply:
I think this should now handle 0 correctly.